substrate/frame/support/src/traits/tokens/fungible/union_of.rs

```diff
@@ -871,6 +871,26 @@ impl<
 		}
 	}
 }
+impl<
+	Reason,
+	Balance,
+	Left: fungible::hold::DoneSlash<Reason, AccountId, Balance>,
+	Right: fungibles::hold::DoneSlash<Right::AssetId, Reason, AccountId, Balance> + fungibles::Inspect<AccountId>,
+	Criterion: Convert<AssetKind, Either<(), Right::AssetId>>,
+	AssetKind: AssetId,
+	AccountId,
+> fungibles::hold::DoneSlash<AssetKind, Reason, AccountId, Balance> for UnionOf<Left, Right, Criterion, AssetKind, AccountId> {
```

Comment:
In which context is this `UnionOf` used?

Reply:
Not sure I understand the question.
UnionOf is used for combining 2 `fungible` and `fungibles` interfaces into one.
Before, it used to implement `slash()` and `done_slash()` as part of the `Balanced` trait, but now since `done_slash()` is in it's own trait, I should implement it for UnionOf separately.

Reply:
My question was a bit "lazy", in that I merely asked in which context this type is useful, instead of looking it up myself :) For example:

> UnionOf is used for combining 2 fungible and fungibles interfaces into one.


When and why would someone need this. 

Anyways, not quite relevant.